### PR TITLE
HTTP/2fy request-upload test around streaming

### PR DESCRIPTION
**Pull request by yoichio**
That's because wpt h1 server doesn't support chunked encoding but h2 supports it natively.
Related thread: https://github.com/whatwg/fetch/issues/966

### CODE REVIEW

File: fetch/api/basic/request-upload.h2.any.js
Comment (yutakahirano) [inline fetch/api/basic/request-upload.h2.any.js:17]: Let's flatten one nesting level.

```
return fetch(...).then((resp) => {
  return resp.text();
}).then((text) => {
  assert_equals(...);
});
```

File: fetch/api/basic/request-upload.h2.any.js
Comment (yutakahirano) [inline fetch/api/basic/request-upload.h2.any.js:7]: [optional] You can use `{method: method}` or `{method}`. Ditto below.

**Comment by foolip** [comment]
The wpt-chrome-dev-* CI failures here were likely due to https://github.com/web-platform-tests/wpt/pull/19360. A workaround was landed in https://github.com/web-platform-tests/wpt/pull/28238, so I'm closing and reopening to re-trigger the CI checks.

**Comment by annevk** [comment]
I guess we haven't quite resolved whether to support H/1, but ideally we have some kind of test for that as well.

**Comment by yoichio** [comment]
> I guess we haven't quite resolved whether to support H/1, 

Yes but supporting H/2 has been decided. Should we also have a wpt that confirms the feature is not supported over H/1?

**Comment by yoichio** [comment]
Undid confirming failure tests, which required wptserve/wptserve/server.py change. Will do later-on.

**Comment by yoichio** [comment]
Ping?

**Review by yutakahirano — APPROVED** [review]
LGTM.

**Comment by annevk** [comment]
@yoichio I'm not sure I understand your comment above. Has there been a decision to only support H/2 and up? If so, a test or a few tests for non-support for H/1 would be great.

**Comment by yoichio** [comment]
> Has there been a decision to only support H/2 and up?
No to "only" but yes to supporting at least H/2.

> If so, a test or a few tests for non-support for H/1 would be great.
That's bit confusing. Should we put a wpt test that points an API doesn't work? This is different from how an API raises an exception.

**Comment by yoichio** [comment]
I'm O.K to just put same tests both in request-upload.any.js and request-upload.h2.any.js

**Comment by annevk** [comment]
All I'm saying is that once we make a decision on what to do for H/1 we should have test coverage for that decision.

**Comment by yoichio** [comment]
Sure.

**Comment by yoichio** [comment]
Ping?

**Comment by annevk** [comment]
@yoichio you should feel free to merge this. Or do you want another review?

**Comment by yoichio** [comment]
I don't have a write access to this repository.
Could you give me the right?

**Comment by annevk** [comment]
(I've asked on W3C IRC `#testing` for someone to give you access.)

**Pull request merged**
